frontend/src/app/services/event-list.ts

```diff
@@ -1,49 +1,71 @@
 import {DayState} from './simulation';
 import {EventTrigger} from './events';
 import {dateDiff} from './utils';
-import {formatNumber} from '../utils/format';
+import _ from 'lodash';
 
-function seasonalDateDiff(date: string, mmdd: string) {
-  const date2 = date.slice(0, 5) + mmdd;
-  return dateDiff(date, date2);
+/**
+ * Generate first true randomly between given dates
+ * TODO: unit test
+ * @param date - simulation date
+ * @param dateFrom - interval start (YYYY-MM-DD)
+ * @param dateTo - interval end (YYYY-MM-DD)
+ */
+function dateBetweenTrigger(date: string, dateFrom: string, dateTo: string) {
+  if (dateDiff(date, dateFrom) < 0) {
+    // before given interval
+    return false;
+  } else if (dateDiff(date, dateTo) < 0) {
+    // in between interval
+    return (1 / (-dateDiff(date, dateTo))) > Math.random();
```

Comment:
I think we should

```suggestion
    return 1 / (1 - dateDiff(date, dateTo)) > Math.random();
```

for `dateDiff(date, dateTo) <= 0`, and `false` otherwise.